ddev/ddev
# feat: get completion for composer commands

Figured I'd get this done while autocompletion is on my brain.

## The Issue

There's no autocompletion for composer commands right now, other than what cobra can give us.

## How This PR Solves The Issue

If the cwd is inside a project with a running web container, we request completions from composer in the container.

This should work for all shells cobra supports autocompletion for - which is more than composer itself!

I've also added autocompletion for the `create` command (pretty much only useful for flags, but that's not nothing) and *hidden* the `create-project` command from completion suggestions.

## Manual Testing Instructions

Running in any shell cobra supports for autocompletion, type `ddev composer <tab>`, and other combinations you'd expect to get autocompletions for (like `ddev composer re<tab>` to get completion suggestions for `composer remove` and `composer require`)

Also try out `ddev composer create --<tab>` to see completion suggestions for the create command.

## Automated Testing Overview



Added a couple of small tests to check we do in fact get expected completions from composer. That test is pretty light because we don't need to actually test composer's auto completion suggestions - we just need to validate that we're passing the completion request through, and getting a correct response.

## Related Issue Link(s)

N/A

## Release/Deployment Notes


Should just work out of the box, no breaking changes, just nice auto complete goodness.
